exercises/go-counting/Example.cs

```diff
@@ -0,0 +1,148 @@
+﻿using System;
+using System.Drawing;
+using System.Collections.Generic;
+using System.Linq;
+
+public class GoCounting
+{
+    public enum Player
+    {
+        None,
+        Black,
+        White
+    }
+
+    private readonly Player[][] board;
+
+    public GoCounting(string input)
+    {
+        board = ParseBoard(input);
+    }
+
+    private static Player CharToPlayer(char c)
+    {
+        switch (c)
+        {
+            case 'B': return Player.Black;
+            case 'W': return Player.White;
+            default: return Player.None;
+        }
+    }
+
+    private static Player[][] ParseBoard(string input)
+    {
+        var split = input.Split('\n');
+        var rows = split.Length;
+        var cols = split[0].Length;
+
+        return split.Select(row => row.Select(CharToPlayer).ToArray()).ToArray();
+    }
+
+    private int Cols => board[0].Length;
+    private int Rows => board.Length;
+
+    private bool IsValidCoordinate(Point coordinate) =>
+        coordinate.Y >= 0 && coordinate.Y < Rows &&
+        coordinate.X >= 0 && coordinate.X < Cols;
+
+    private Player GetPlayer(Point coordinate) => board[coordinate.Y][coordinate.X];
+
+    private bool isEmpty(Point coordinate) => GetPlayer(coordinate) == Player.None;
+    private bool isTaken(Point coordinate) => !isEmpty(coordinate);
+
+    private IEnumerable<Point> EmptyCoordinates()
+    {
+        return Enumerable.Range(0, Cols).SelectMany(col =>
+               Enumerable.Range(0, Rows).Select(row => new Point(col, row)))
+                .Where(isEmpty);
+    }
+
+    private IEnumerable<Point> NeighborCoordinates(Point coordinate)
+    {
+        var row = coordinate.Y;
+        var col = coordinate.X;
+
+        var coords = new[]
+        {
+            new Point(col, row - 1),
+            new Point(col-1, row),
+            new Point(col+1, row),
+            new Point(col, row+1)
+        };
+
+        return coords.Where(IsValidCoordinate);
+    }
+
+    private IEnumerable<Point> NonEmptyNeighborCoordinates(Point coordinate) =>
```

Comment:
Great point! I've updated the code and also renamed `isTaken` and `isEmpty` to `IsTaken` and `IsEmpty` and then used `IsTaken` in `TakenNeighborCoordinates` and removed the unnecessary call to `IsTaken` in `TerritoryOwner`.